src/RustyObjectStore.jl

```diff
@@ -664,28 +708,32 @@ Fetches the data bytes at `path` and writes them to the given `buffer`.
 function get_object!(buffer::AbstractVector{UInt8}, path::String, conf::AbstractConfig)
     response = Response()
     size = length(buffer)
-    cond = Base.AsyncCondition()
-    cond_handle = cond.handle
+    ct = current_task()
+    event = Base.Event()
+    handle = pointer_from_objref(event)
     config = into_config(conf)
     while true
-        result = GC.@preserve buffer config response cond begin
+        preserve_task(ct)
```

Comment:
I don't understand this. Why would we need to GC-preserve our own task, if we're also _blocking_, waiting for the event to finish? Of course the task will be kept alive since the task itself is waiting on the event. ...

... oh, is the issue that the `Event` itself is only rooted from our own stack, and the task is only rooted by the Event's wait-queue, so it's a cycle and that cycle can get GC'd!? 😅 😅 😅 Wow, that's wild.

I think that this deserves some comments explaining this, and explaining why the preserve is needed.

Reply:
Also, why couldn't this use `Base.preserve_handle(ct)`? Is it to reduce contention?